selfdrive/ui/qt/onroad/annotated_camera.cc

```diff
@@ -23,6 +24,12 @@ AnnotatedCameraWidget::AnnotatedCameraWidget(VisionStreamType type, QWidget* par
   dm_img = loadPixmap("../assets/img_driver_face.png", {img_size + 5, img_size + 5});
 }
 
+// Checks if "Always-On Driver Monitoring" toggle is enabled in settings
+bool AnnotatedCameraWidget::isDMAlwaysOn() const {
+  Params params;
+  return params.getBool("AlwaysOnDM");
+}
+
```

Comment:
`params.get()` is expensive. Move parameter reading to `void AnnotatedCameraWidget::showEvent(QShowEvent *event)` to update it once when the widget becomes visible, instead of repeating it on every update.

Reply:
I went with FrogAi's solution. Thanks for pointing this out in the first place